pkg/common/consumer/consumergroup_offsets_checker.go

```diff
@@ -0,0 +1,74 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/eventing-kafka/pkg/common/kafka/offset"
+)
+
+const (
+	OffsetCheckRetryTimeout  = 60 * time.Second
+	OffsetCheckRetryInterval = 5 * time.Second
+)
+
+// wrapper functions for the Sarama functions, to facilitate unit testing
+var newSaramaClient = sarama.NewClient
+var newSaramaClusterAdmin = sarama.NewClusterAdmin
+
+type ConsumerGroupOffsetsChecker interface {
+	WaitForOffsetsInitialization(ctx context.Context, groupID string, topics []string, logger *zap.SugaredLogger, addrs []string, config *sarama.Config) error
+}
+
+type NoopConsumerGroupOffsetsChecker struct {
+}
+
+func (c *NoopConsumerGroupOffsetsChecker) WaitForOffsetsInitialization(ctx context.Context, groupID string, topics []string, logger *zap.SugaredLogger, addrs []string, config *sarama.Config) error {
+	return nil
+}
+
+type KafkaConsumerGroupOffsetsChecker struct {
+}
+
+func (k *KafkaConsumerGroupOffsetsChecker) WaitForOffsetsInitialization(ctx context.Context, groupID string, topics []string, logger *zap.SugaredLogger, addrs []string, config *sarama.Config) error {
+	logger.Infow("Checking if all offsets are initialized", zap.Any("topics", topics), zap.Any("groupID", groupID))
+
+	client, err := newSaramaClient(addrs, config)
+	if err != nil {
+		logger.Errorw("Unable to create Kafka client", zap.Error(err))
```

Comment:
1. NIT error message starting with lower case
1. would be helpful to add the groupID and topics

Reply:
> NIT error message starting with lower case

the `Unable to create Kafka client` is starting with upper case... which is correct? :thinking: